### ISSUE TITLE
Special characters break parsing

### ISSUE THREAD

**Issue by TripleDogDare** [knsv/mermaid#54]
Currently Mermaid does not handle special characters inside elements.
Not only is it not allowed, but it also breaks the graph.

Building on a simple demo
```
graph LR;
    A[Object]-->B(Thing);
```
I'd like to use something with parenthesis in the text of an element.
```
graph LR;
    A[Object(foo,bar)]-->B(Thing);
```
I also attempted to use HTML character codes with no result.
```
graph LR;
    A[Object&#40;foo,bar&#41;]-->B(Thing);
```

**Comment by timelyportfolio** [comment]
I believe this is related, so I'll file under this issue, but happy to separate into its own issue if that is better.  So far, it seems `style` and `%` breaks the style parser, so something like

**not working**
```
graph TD;
     classDef exClass font-style:bold;
     a-->b;
     class a exClass;

graph TD;
     classDef exClass font-size:50%;
     a-->b;
     class a exClass;
```
**working**
```
graph TD;
     classDef exClass font-family:monospace;
     a-->b;
     class a exClass;

graph TD;
     classDef exClass font-size:10px;
     a-->b;
     class a exClass;
```

**Comment by knsv** [comment]
Related it is! Will take a look.

**Comment by knsv** [comment]
A fix is committed For the style/% problem. Will be included in the next release.

**Comment by Xtof-** [comment]
Waiting special character implementation, I found this alternative to simulate bracket : "<." and ".>".

**Comment by realchaseadams** [comment]
@knsv in the example: 

```
graph TD
question1{Gas tank less than 1/8?}
action1[Fill tank to 100%]
question1-- Yes -->action1
```

An error is thrown, very similar to @Tripledogdare, maybe there should be parsing logic to handle html entities altogether?

**Comment by knsv** [comment]
Hi Chase! Thanks for bringing this up. I expect there several other that are experiencing the same thing. I have added % as a valid text character so with the next release this will work. Perhaps we should sum thing up and do bigger fix for the characters. Right now its one at a time when they pop up.

Anyways, with 0.3.3 this will work

**Comment by realchaseadams** [comment]
@knsv great turn around time! I agree, it'd be a really good idea to gather up all special characters that are processed as tags and make sure they're handled properly when it's appropriate.

This is a really cool product, nice work!

**Comment by knsv** [comment]
I am going through issues and am closing this one. I think special characters are now much improved with the fix allowing any characters within quotes (#162).

**Comment by The-Best-Codes** [comment]
Something like this:

```
graph TD
    1 --> email@example.com
```

Breaks due to the `@` symbol. I'd like to be able to put emails…
